File: src/Other/RemoveElement.java
package Other;

import java.util.Arrays;

// https://leetcode.com/problems/remove-element/
public class RemoveElement {
    public int removeElement(int[] nums, int val) {
        int slow = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) {
                nums[slow++] = nums[i];
            }
        }

        return slow;
    }

    public static void main(String[] args) {
        int[] nums = {0,1,2,2,3,0,4,2};
        RemoveElement r = new RemoveElement();
        System.out.println(r.removeElement(nums,2));
        System.out.println(Arrays.toString(nums));
    }
}

File: py/essential/RemoveElement.py
# https://leetcode.com/problems/remove-element/description/
# Neet

from typing import List


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        """
        Do not return anything, modify nums in-place instead.
        """
        i = 0 # left
        for j in range(len(nums)):
            if nums[j] != val:
                nums[i] = nums[j]
                i += 1
        return i

s = Solution()
print(s.removeElement([3, 2, 2, 3], 3))
input = [0, 1, 2, 2, 3, 0, 4, 2]
print(s.removeElement(input, 2))
print(input)
print(s.removeElement([3, 3, 3, 3], 3))